arcade/coin/src/main.leo

```diff
@@ -0,0 +1,26 @@
+program puzzle_arcade_coin_v002.aleo {
+  record PuzzleArcadeCoin {
+    owner: address,
+    amount: u64
+  }
+
+  transition mint(to: address) -> PuzzleArcadeCoin {
+    assert_eq(self.caller, self.signer);
+    assert_eq(self.caller, aleo1pedhtu6akw9z68wedu8t3fgxfdh3ye2rypeqkx9cxjpr99chqvyqkjg7rw);
+
+    let coin: PuzzleArcadeCoin = PuzzleArcadeCoin {
+      owner: to,
+      amount: 1000000u64
+    };
+
+    return (coin);
+  }
+
+  transition spend(coin: PuzzleArcadeCoin, amount: u64) -> PuzzleArcadeCoin {
```

Comment:
this is serving as a burn function, right? thinking about this some more there's no "burn address" concept like in evm land? https://etherscan.io/address/0x000000000000000000000000000000000000dead

just thinking about record mgmt here and overhead there